shell/src/main/java/org/apache/accumulo/shell/commands/ListScansCommand.java

```diff
@@ -75,14 +77,31 @@ public int numArgs() {
   public Options getOptions() {
     final Options opts = new Options();
 
-    tserverOption = new Option("ts", "tabletServer", true, "tablet server to list scans for");
-    tserverOption.setArgName("tablet server");
-    opts.addOption(tserverOption);
+    serverOpt = new Option("s", "server", true, "tablet/scan server regex to list scans for");
+    serverOpt.setArgName("tablet/scan server regex");
+    opts.addOption(serverOpt);
+
+    rgOpt = new Option("rg", "resourceGroup", true,
+        "tablet/scan server resource group regex to list scans for");
+    rgOpt.setArgName("resource group");
+    opts.addOption(rgOpt);
 
     disablePaginationOpt = new Option("np", "no-pagination", false, "disable pagination of output");
     opts.addOption(disablePaginationOpt);
 
     return opts;
   }
 
+  static BiPredicate<String,Integer> serverRegexPredicate(CommandLine cl, Option serverOpt) {
+    return Optional.ofNullable(cl.getOptionValue(serverOpt))
+        .map(regex -> (BiPredicate<String,
+            Integer>) (h, p) -> Pattern.compile(regex).matcher(h + ":" + p).matches())
+        .orElse((h, p) -> true);
```

Comment:
This seemed like its compiling the regex for each call to the predicate.  Was wondering if this could be avoided experimented locally and came up with the following.  I did not know about `.asMatchPredicate()` until seeing it a few lines down, that is really neat.

```suggestion
    return Optional.ofNullable(cl.getOptionValue(serverOpt)).map(regex->Pattern.compile(regex).asMatchPredicate())
            .map(matcherPredicate -> (BiPredicate<String, Integer>) (h, p) -> matcherPredicate.test(h+":"+p)).orElse((h, p) -> true);
```